### PR TITLE
Support the devicePreference

### PR DESCRIPTION
**Pull request by lionkunonly**
Support the devicePreference for `MLContext`.

### CODE REVIEW

File: node/src/Context.cpp [hunk 1]
```diff
@@ -30,18 +30,33 @@ namespace node {
             if (optionsObject.Has("powerPreference")) {
                 std::string powerPreferenceOption = optionsObject.Get("powerPreference").ToString();
```
Comment (lionkunonly) [inline node/src/Context.cpp:31]: Should I check if the content of the `powerPreference` is a String before I convert it ?

File: node/src/Context.cpp [hunk 2]
```diff
@@ -30,18 +30,33 @@ namespace node {
...
+                } else {
+                    Napi::Error::New(info.Env(), "Invaild PowerPreference")
+                        .ThrowAsJavaScriptException();
+                    return;
+                }
+            }
+
+            if (optionsObject.Has("devicePreference")) {
+                std::string devicePreferenceOption =
+                    optionsObject.Get("devicePreference").ToString();
```
Comment (lionkunonly) [inline node/src/Context.cpp:47]: The same question.

File: node/src/Context.cpp (hunk 1 above)
Comment (fujunwei) [reply node/src/Context.cpp:31]: Yes, please add the condition.

Comment (lionkunonly) [reply node/src/Context.cpp:36]: I have added the condition. PTAL.

**Review by fujunwei — APPROVED** [review]
LGTM

**Review by huningxin — APPROVED** [review]

**Pull request merged**